core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -637,6 +662,24 @@ private[spark] class ExecutorAllocationManager(
     def isExecutorIdle(executorId: String): Boolean = {
       !executorIdToTaskIds.contains(executorId)
     }
+
+    /**
+     * Get a tuple of (the number of task with locality preferences, a map where each pair is a
+     * node and the number of tasks that would like to be scheduled on that node).
+     */
```

Comment:
Mention "These hints are updated when stages arrive and complete, so are not up-to-date at task granularity within stages."